### ISSUE TITLE
Offline use of genome properties

### ISSUE THREAD

**Issue by SilasK** [ebi-pf-team/genome-properties#44]
Hello the genome properties are really great. 

I wonder if it is possible to use them offline in a command line script or so. Could you point me to which pearl script I have to use?

**Comment by LornaMGnify** [comment]
Hi Silas,
Thanks for getting in touch. We just updated the website with some information on running a Genome Properties analysis locally. Have a look at the page linked below (do a hard refresh if you have visited the site before), hopefully it will give you the information you need. 
If you still need more help or information, please do get back in touch.
Thanks,
Lorna.

https://www.ebi.ac.uk/interpro/genomeproperties/#calculating

**Comment by SilasK** [comment]
ok found it finaly https://wwwdev.ebi.ac.uk/interpro/genomeproperties/#calculating

**Comment by LornaMGnify** [comment]
If you use the www site rather than the wwwdev, you will get the most up to date public site. The wwwdev may appear and dissapear as we make code changes, so it not recommended.

**Comment by SilasK** [comment]
Thank you @happy-lorna 

Do you have any suggestion how to install all the perl dependencies? I'm trying to install them either on my mac or on the linux server without success.

**Comment by LornaMGnify** [comment]
Did you set the PERL5LIB? e.g.
PERL5LIB=$PERL5LIB:<path to GP>/code/modules

**Comment by SilasK** [comment]
Yes, I’m struggling with the Data::Printer package and all its dependencies

**Comment by LornaMGnify** [comment]
OK in that case I'm afraid you might have to wait until @rdfinn can respond - he wrote all the code and so can answer your question much better that I can. sorry!

**Comment by SilasK** [comment]
I found the docker file in https://github.com/Micromeda/genome-properties
This works for me,

May be you want to fork it back.

**Comment by LornaMGnify** [comment]
You should be aware that there have been quite a number of changes to this code since this fork. Having spoken to @rdfinn he said you can just comment out the Data::Printer line anyway, as it is a hangover from bug fixing. 
If you are still having problems he can probably reply next week, but not before then.

**Comment by LeeBergstrand** [comment]
@SilasK @happy-lorna @rdfinn I have made a pull request with my docker implementation, test, ci etc from my fork.

**Comment by SilasK** [comment]
I try to make a conda package for genome-properties. But I don't find a way to install the following Perl packages: 
`LWP::UserAgent` which depends on `Data::Printer`

In addition, would it be possible to set defaults for the options `-gpdir -gpff`to something meaning full?

**Comment by LeeBergstrand** [comment]
@SilasK https://anaconda.org/bioconda/perl-app-cpanminus

In branch, https://github.com/ebi-pf-team/genome-properties/tree/rel3.0

I've had some of my fork's code merged in. It uses cpanm to install the packages during travis ci installs.

**Comment by LeeBergstrand** [comment]
Travis-CI file.
```
language: perl
perl:
- "5.26"
- "5.24"
- "5.22"
- "5.20"
install:
  - cpanm --installdeps ./code/
  - export PERL5LIB=./code/modules
script:
  - perl ./code/scripts/assign_genome_properties.pl -all -gpdir ./flatfiles -gpff genomeProperties.txt -matches testing/testfiles/ecoli_k12_annotation.tsv -outdir ./ -outfiles summary -name ecoli_k12
  - test -s ./SUMMARY_FILE_ecoli_k12
```

**Comment by LeeBergstrand** [comment]
@SilasK https://github.com/ebi-pf-team/genome-properties/blob/rel3.0/code/cpanfile

This is the perl equivelent to python's ```requirements.txt``` and you use ```cpanm``` like you would use ```pip```.

https://anaconda.org/bioconda/perl-app-cpanminus
